packages/dredd-transactions/scripts/prepack.js

```diff
@@ -4,14 +4,60 @@
 
 const fs = require('fs');
 const path = require('path');
+const rimraf = require('rimraf');
 
-const drafterPackageData = require('../../../node_modules/drafter/package');
+const PACKAGE_DIR = path.resolve(__dirname, '..');
+const SYMLINKS_LOG = path.join(PACKAGE_DIR, 'prepack-symlinks.log');
+const DRAFTER_PATH = path.join(PACKAGE_DIR, 'node_modules', 'drafter');
 
+
+function readPackageJson(packageDir) {
+  return JSON.parse(fs.readFileSync(path.join(packageDir, 'package.json')));
+}
+
+function writePackageJson(packageDir, packageData) {
+  fs.writeFileSync(path.join(packageDir, 'package.json'), JSON.stringify(packageData, null, 2));
+}
+
+/**
+ * Goes through the whole dependency tree of a given top-level dependency
+ * and makes sure all packages involved are accessible in the local
+ * 'node_modules' directory at least in form of symlinks.
+ *
+ * This is useful if the project uses yarn workspaces, but wants to publish
+ * the package with npm and uses 'bundledDependencies'. 'npm pack' is not able
+ * to find packages if they're put into the root 'node_modules' by yarn and
+ * wouldn't be able to bundle them.
+ *
+ * @param {string} dependencyName Dependency tree root
+ */
+function symlinkDependencyTreeToLocalNodeModules(dependencyName) {
+  const localDependencyPath = path.join(PACKAGE_DIR, 'node_modules', dependencyName);
+  if (!fs.existsSync(localDependencyPath)) {
+    fs.symlinkSync(`../../../node_modules/${dependencyName}`, localDependencyPath);
```

Comment:
Maybe it's better to get the root of the project here, instead of doing `../../..` in case we change the directories later.

Reply:
Since `dredd-transactions` package is always the direct child of the `dredd/packages`, the only thing that is variable is the location of the `prepack.js` script, which casts module requires.

I don't think it's worthy to invest time into this, as `scripts` are purely technical, and touching them should raise expectations that things may break.

Reply:
Actually, the first argument of `fs.symlinkSync()` must be a relative path from the resulting symlink's location:

<img width="950" alt="image" src="https://user-images.githubusercontent.com/283441/67218049-01f0be80-f426-11e9-81be-6c6f34b403bb.png">


https://nodejs.org/api/fs.html#fs_fs_symlink_target_path_type_callback

I wouldn't be too concerned about this as if things change, the smoke test is our insurance.